File: CS/RestrictedDraggingExample/Docking/CustomDockPanel.cs
using DevExpress.XtraBars.Docking;
using DevExpress.XtraBars.Docking.Helpers;

namespace RestrictedDraggingExample.Docking {
    public class CustomDockPanel :DockPanel {
        public CustomDockPanel(DockManager manager)
            : base(true, DockingStyle.Float, manager) { }

        protected override DockPanelMouseHandler CreateMouseHandler() {
            return new CustomDockPanelMouseHandler(this);
        }
    }
}

File: VB/RestrictedDraggingExample/Docking/CustomDockPanel.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraBars.Docking
Imports DevExpress.XtraBars.Docking.Helpers

Namespace RestrictedDraggingExample.Docking
	Public Class CustomDockPanel
		Inherits DockPanel
		Public Sub New(ByVal manager As DockManager)
			MyBase.New(True, DockingStyle.Float, manager)
		End Sub

		Protected Overrides Function CreateMouseHandler() As DockPanelMouseHandler
			Return New CustomDockPanelMouseHandler(Me)
		End Function
	End Class
End Namespace
